ambari/src/main/java/io/brooklyn/ambari/AmbariClusterImpl.java

```diff
@@ -509,7 +509,7 @@ private RecommendationWrapper getRecommendationWrapperFromAmbariHostGroups() {
                 newConfigurationMap.put(stringMapEntry.getKey(), stringMapEntry.getValue());
             } else {
                 if(stringMapEntry.getValue() != null) {
-                    newConfigurationMap.get(stringMapEntry.getKey()).putAll(stringMapEntry.getValue());
+                    newConfigurationMap.putAll(stringMapEntry.getValue());
```

Comment:
It does not make sense to me because you're merging the value map one level up. Let me explain with an example: Let say `configuration` looks like this:
```json
{
  "key1": {
    "a": "b",
    "c": "d"
  }
}
```

and `serviceConfig` looks like this:
```json
{
  "key1": {
    "e": "j"
  }
}
```

Your code will merge the two as follow (which should fails because the value is now a `String` instead of `Map`)
```json
{
  "key1": {
    "a": "b",
    "c": "d"
  },
  "e": "j"
}
```
whereas the old code does this:
```json
{
  "key1": {
    "a": "b",
    "c": "d",
    "e": "j"
  }
}
```